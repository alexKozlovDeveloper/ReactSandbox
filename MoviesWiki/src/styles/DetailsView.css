.body {
    background-color: tomato;
    display: flex;
    padding-left: 30px;
    padding-right: 30px;
    padding-top: 10px;
    padding-bottom: 10px;
}

.imagecontainer {
    min-width: 150px;
}

.detailscontainer {
    padding: 10px;
}

.maintitlecontainer {
    display: flex;
}

.propertiescontainer {
    display: flex;
}

.descriptioncontainer {
    
}