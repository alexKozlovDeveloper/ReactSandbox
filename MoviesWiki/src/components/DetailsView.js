import React, { Component } from "react";
import { connect } from 'react-redux'

import Image from "./Common/Image";
import DetailsTitle from "./Common/DetailsTitle";
import Rating from "./Common/Rating";
import Description from "./Common/Description";
import YearAndDuration from "./Common/YearAndDuration";
import BackButton from "./Common/BackButton";
import DetailsViewNoItemLabel from "./Common/DetailsViewNoItemLabel";

import styles from "../styles/DetailsView.css"


class DetailsView extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        var content;

        if (this.props.item != null) {
            content = (<>
                <div className={styles.imagecontainer}>
                    <Image image={this.props.item.poster_path} />
                </div>
                <div className={styles.detailscontainer}>
                    <div className={styles.maintitlecontainer}>
                        <DetailsTitle title={this.props.item.title} />
                        <Rating rating={this.props.item.vote_average} />
                    </div>
                    <div className={styles.propertiescontainer}>
                        <YearAndDuration year={this.props.item.release_date} duration={this.props.item.runtime} />
                    </div>
                    <div className={styles.descriptioncontainer}>
                        <Description description={this.props.item.overview} />
                    </div>
                </div>
                <div>
                    <BackButton buttonText="" />
                </div>
            </>)
        } else {
            content = (<DetailsViewNoItemLabel></DetailsViewNoItemLabel>)
        }

        return (
            <div className={styles.body}>
                {content}
            </div>
        );
    }
}

function mapStateToProps(state) {
    const { selectedItem } = state.moviesReducer;

    return {
        item: selectedItem
    };
}

export default connect(mapStateToProps)(DetailsView);
