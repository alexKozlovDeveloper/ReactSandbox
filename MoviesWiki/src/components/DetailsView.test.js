import React from 'react';
import { mount, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import rootReducer from '../reducers';

import DetailsView from './DetailsView';

configure({ adapter: new Adapter() });

test('ItemDetails rendering test', () => {
  const item = {
    title: 'Interstellar 4',
    releaseDate: '2017',
    image: 'https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg',
    genre: 'action 4',
    rating: 7.2,
    duration: 145,
    description: 'is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.',
  };

  const defaultStore = {
    moviesReducer: {
      selectedItem: item,
    },
  };

  const store = createStore(rootReducer, defaultStore);

  const component = mount(<Provider store={store}>
        <DetailsView item={item} />
    </Provider>);
  expect(shallowToJson(component)).toMatchSnapshot();
});

test('ItemDetails noItem rendering test', () => {
  const defaultStore = {
    moviesReducer: {
      selectedItem: null,
    },
  };

  const store = createStore(rootReducer, defaultStore);

  const component = mount(<Provider store={store}>
        <DetailsView />
    </Provider>);
  expect(shallowToJson(component)).toMatchSnapshot();
});
