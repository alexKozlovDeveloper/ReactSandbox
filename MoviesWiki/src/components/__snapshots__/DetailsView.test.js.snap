// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetails noItem rendering test 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(DetailsView)>
    <DetailsView
      dispatch={[Function]}
      item={null}
    >
      <div
        className="body"
      >
        <DetailsViewNoItemLabel>
          <div
            className="title"
          >
            Select any movie...
          </div>
        </DetailsViewNoItemLabel>
      </div>
    </DetailsView>
  </Connect(DetailsView)>
</Provider>
`;

exports[`ItemDetails rendering test 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(DetailsView)
    item={
      Object {
        "description": "is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.",
        "duration": 145,
        "genre": "action 4",
        "image": "https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg",
        "rating": 7.2,
        "releaseDate": "2017",
        "title": "Interstellar 4",
      }
    }
  >
    <DetailsView
      dispatch={[Function]}
      item={null}
    >
      <div
        className="body"
      >
        <DetailsViewNoItemLabel>
          <div
            className="title"
          >
            Select any movie...
          </div>
        </DetailsViewNoItemLabel>
      </div>
    </DetailsView>
  </Connect(DetailsView)>
</Provider>
`;
